### PR TITLE
Resolves remaining model conflicts

### PR DESCRIPTION
**Pull request by Rotorsoft**
See ticket

## Link to Issue
Closes: #7575

## Description of Changes
- Deprecates associate hook
- Refactors association utils
- Fixes test and typing issues
- Removes orphan reactions - pointing to invalid address_id

### CODE REVIEW

File: packages/commonwealth/server/migrations/20240502120431-reconcile-xrefs.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+alter table only public."Reactions" 
+add constraint "Reactions_address_id_fkey"
+foreign key (address_id) references public."Addresses"(id);
+
+-- Many-to-many (X-ref) pattern with composite PK
+
+ALTER TABLE ONLY public."ThreadSubscriptions" DROP COLUMN IF EXISTS "id";
+ALTER TABLE ONLY public."ThreadSubscriptions"
+ADD CONSTRAINT "Threadsubscriptions_pkey" PRIMARY KEY (thread_id, user_id);
```
Comment (timolegros) [inline packages/commonwealth/server/migrations/20240502120431-reconcile-xrefs.js:20]: Might as well remove `id` from `CommunityAlerts` too...

**Review by timolegros — CHANGES_REQUESTED** [review]

File: packages/commonwealth/server/migrations/20240502120431-reconcile-xrefs.js (hunk 1 above)
Comment (Rotorsoft) [reply packages/commonwealth/server/migrations/20240502120431-reconcile-xrefs.js:20]: Merging master 😱 ... removing snapshot tables and making community alerts an x-ref in next commit.

**Comment by Rotorsoft** [comment]
> Big update. Is there any recommended test plan? Should we have QA give it a pass and/or are the automated tests able to verify DB model correctness against the schemas?

@ilijabojanovic is PTO for a few days, so we must try to play with the main flows to be sure... all automated tests are passing

**Review by rbennettcw — APPROVED** [review]
Seems to work

**Review by timolegros — APPROVED** [review]

**Pull request merged**